### PR TITLE
add JWT authorization

### PR DESCRIPTION
**Pull request by lezrak**
JWT authorization added, post username and password in request body at /login and you will receive access token valid for 10 days, add it as Bearer token to your requests to consume API. 
No refresh tokens deemed necessary at this point.

### CODE REVIEW

File: src/main/java/pl/neurocars/backend/BackendApplication.java [hunk 1]
```diff
@@ -16,12 +15,7 @@ public static void main(String[] args) {
...
 
     @Bean
-    public WebMvcConfigurer corsConfigurer() {
-        return new WebMvcConfigurer() {
-            @Override
-            public void addCorsMappings(CorsRegistry registry) {
-                registry.addMapping("/**").allowedMethods("*");//.allowedOrigins("https://neurocars.herokuapp.com", "https://neurocars-learning.herokuapp.com", "http://localhost:3000");
-            }
-        };
+    public BCryptPasswordEncoder bCryptPasswordEncoder() {
```
Comment (BartoszF) [inline src/main/java/pl/neurocars/backend/BackendApplication.java]: It should go to SecurityConfig. Leave main class as lean as possible

File: src/main/java/pl/neurocars/backend/security/JWTAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                                                HttpServletResponse res) throws AuthenticationException {
+        try {
+            PlayerDTO creds = new ObjectMapper()
+                    .readValue(req.getInputStream(), PlayerDTO.class);
+
+            return authenticationManager.authenticate(
+                    new UsernamePasswordAuthenticationToken(
+                            creds.getUsername(),
+                            creds.getPassword(),
+                            new ArrayList<>())
```
Comment (BartoszF) [inline src/main/java/pl/neurocars/backend/security/JWTAuthenticationFilter.java]: We need to think, if we need roles for users

File: src/main/java/pl/neurocars/backend/security/SecurityConstants.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package pl.neurocars.backend.security;
+
+public class SecurityConstants {
```
Comment (BartoszF) [inline src/main/java/pl/neurocars/backend/security/SecurityConstants.java]: These things should be configurable from properties

File: src/main/java/pl/neurocars/backend/security/SecurityConstants.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    public static final String SECRET = "AfterDevelopmentPhaseThisShouldBeEnvVariable";
+    public static final long EXPIRATION_TIME = 864_000_000; // 10 days
```
Comment (BartoszF) [inline src/main/java/pl/neurocars/backend/security/SecurityConstants.java]: 10 days? Isn't it too long?

File: src/main/java/pl/neurocars/backend/BackendApplication.java [hunk 5]
```diff
@@ -16,12 +15,7 @@ public static void main(String[] args) {
 	}
 
     @Bean
-    public WebMvcConfigurer corsConfigurer() {
```
Comment (BartoszF) [inline src/main/java/pl/neurocars/backend/BackendApplication.java]: And where it is now?

Comment (lezrak) [reply src/main/java/pl/neurocars/backend/BackendApplication.java]: SecurityConfig:31 
http.cors().and().csrf().disable().authorizeRequests()
I believe it does the job.

File: src/main/java/pl/neurocars/backend/security/SecurityConstants.java (hunk 4 above)
Comment (lezrak) [reply src/main/java/pl/neurocars/backend/security/SecurityConstants.java]: We aren't really afraid of our tokens being stolen at the moment, we just want to have security properly configured. What we don't want is to get unathorized in random moments while still in early development phase. Thus I believe 10 days is ok for now, but if you prefer any other value let me know and I will change it.

File: src/main/java/pl/neurocars/backend/security/SecurityConstants.java (hunk 3 above)
Comment (lezrak) [reply src/main/java/pl/neurocars/backend/security/SecurityConstants.java]: Application properties or environmental variables?

File: src/main/java/pl/neurocars/backend/security/JWTAuthenticationFilter.java (hunk 2 above)
Comment (lezrak) [reply src/main/java/pl/neurocars/backend/security/JWTAuthenticationFilter.java]: As this wasn't discussed before I don't think it should be consider as blocker for this pull, rather as topic to keep in mind in future.

File: src/main/java/pl/neurocars/backend/BackendApplication.java (hunk 1 above)
Comment (lezrak) [reply src/main/java/pl/neurocars/backend/BackendApplication.java]: You can't Autowire Bean in class in which it's declared.

File: src/main/java/pl/neurocars/backend/security/SecurityConstants.java (hunk 3 above)
Comment (BartoszF) [reply src/main/java/pl/neurocars/backend/security/SecurityConstants.java]: app

File: src/main/java/pl/neurocars/backend/BackendApplication.java (hunk 5 above)
Comment (BartoszF) [reply src/main/java/pl/neurocars/backend/BackendApplication.java]: Is it tested?
I think that its enable CORS AND disable CSRF.

Comment (lezrak) [reply src/main/java/pl/neurocars/backend/BackendApplication.java]: CORS configuration fixed.

**Pull request merged**